terraform/main.tf

```diff
@@ -0,0 +1,117 @@
+provider "aws" {
+  region = "us-east-1"
+  profile = "serverless"
+}
+
+data "aws_vpc" "default" {
+  vpc_id = "vpc-0d60ef77"
+}
+
+data "aws_subnet_ids" "all" {
+  vpc_id = data.aws_vpc.default.id
+}
+
+resource "aws_elasticache_cluster" "network-data" {
+  cluster_id           = "network-data"
+  engine               = "redis"
+  node_type            = "cache.t3.small"
+  num_cache_nodes      = 1
+  parameter_group_name = "default.redis6.x"
+  engine_version       = "6.x"
+  port                 = 6379
+}
+
+resource "aws_ecr_repository" "data-collector" {
+  name                 = "data-collector"
+  image_tag_mutability = "MUTABLE"
+}
+
+resource "aws_iam_role" "ec2_role_data_collector" {
+  name = "ec2_role_data_collector"
+
+  assume_role_policy = <<EOF
+{
+  "Version": "2012-10-17",
+  "Statement": [
+    {
+      "Action": "sts:AssumeRole",
+      "Principal": {
+        "Service": "ec2.amazonaws.com"
+      },
+      "Effect": "Allow"
+    }
+  ]
+}
+EOF
+
+  tags = {
+    project = "hello-world"
```

Comment:
We could change to "hathor-explorer-service"

Reply:
Maybe it is also a good idea to include this tag in all other created resources by putting it in the provider: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/guides/resource-tagging#propagating-tags-to-all-resources